test/e2e/examples.go

```diff
@@ -157,6 +160,111 @@ var _ = Describe("Examples e2e", func() {
+			}
+
+			// create two backend services
+			for _, ns := range namespaces {
+				runKubectl("create", "-f", backendSvcYaml, getNsCmdFlag(ns))
+			}
+
+			// wait for objects
+			for _, ns := range namespaces {
+				waitForRCPodsRunning(c, ns.Name, backendRcName)
+				waitForService(c, ns.Name, backendSvcName, true, poll, serviceStartTimeout)
+			}
+			// it is not enough that pods are running because they may be set to running, but
+			// the application itself may have not been initialized. Just query the application.
+			for _, ns := range namespaces {
+				label := labels.SelectorFromSet(labels.Set(map[string]string{"name": backendRcName}))
+				pods, err := c.Pods(ns.Name).List(label, fields.Everything())
+				Expect(err).NotTo(HaveOccurred())
+				err = podsResponding(c, ns.Name, backendPodName, false, pods)
+				Expect(err).NotTo(HaveOccurred())
+				Logf("found %d backend pods responding in namespace %s", len(pods.Items), ns.Name)
+
+				err = serviceResponding(c, ns.Name, backendSvcName)
+				Expect(err).NotTo(HaveOccurred(), "waiting for the service to respond")
+			}
+
+			// Now another tricky part:
+			// It may happen that the service name is not yet in DNS.
+			// So if we start our pod, it will fail. We must make sure
+			// the name is already resolvable. So let's try to query DNS from
+			// the pod we have, until we find our service name.
+			// This complicated code may be removed if the pod itself retried after
+			// dns error or timeout.
+			// This code is probably unnecessary, but let's stay on the safe side.
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"name": backendPodName}))
+			pods, err := c.Pods(namespaces[0].Name).List(label, fields.Everything())
+
+			if err != nil || pods == nil || len(pods.Items) == 0 {
+				Failf("no running pods found")
+			}
+			podName := pods.Items[0].Name
+
+			pythonCode := `
+import socket
+try:
+					socket.gethostbyname('%s')
+					print 'ok'
+except:
+					print 'err'`
+
+			queryDns := fmt.Sprintf(pythonCode, backendSvcName+"."+namespaces[0].Name)
+			_, err = lookForStringInPodExec(namespaces[0].Name, podName, []string{"python", "-c", queryDns}, "ok", dnsReadyTimeout)
+			Expect(err).NotTo(HaveOccurred(), "waiting for output from pod exec")
+
+			updatedPodYaml := prepareResourceWithReplacedString(frontendPodYaml, "dns-backend.development.cluster.local", fmt.Sprintf("dns-backend.%s.cluster.local", namespaces[0].Name))
+
+			// create a pod in each namespace
+			for _, ns := range namespaces {
+				newKubectlCommand("create", "-f", "-", getNsCmdFlag(ns)).withStdinData(updatedPodYaml).exec()
+			}
+			// remember that we cannot wait for the pods to be running because our pods terminate by themselves.
+
+			// wait for pods to print their result
+			for _, ns := range namespaces {
+				_, err := lookForStringInLog(ns.Name, frontendPodName, frontendPodContainerName, podOutput, podStartTimeout)
+				Expect(err).NotTo(HaveOccurred())
+			}
+		})
```

Comment:
namespaces created via createTestingNS are automatically deleted (with everything inside). No need for custom cleanup.